src/extension.ts

```diff
@@ -31,13 +31,18 @@ abstract class Extension {
 	abstract onHotlistPage (): void;
 	abstract onProfilePage (uok: UsernameOrKaid): void;
 	abstract onHomePage (uok: UsernameOrKaid): void;
+	abstract onBadgesPage (url: Array<string>): void | Promise<void>;
 	abstract onPage (): void;
 	abstract onProgram404Page (): void;
 	abstract onDiscussionPage (): void;
 	async init (): Promise<void> {
 		if (window.location.host.includes("khanacademy.org")) {
 			this.onPage();
 
+			if ((this.url[3] === "profile" && this.url[5] === "badges") || this.url[3] === "badges") {
+				this.onBadgesPage(this.url);
```

Comment:
Would prefer to move this to below the conditional statement for `onHomePage()`, but because of the current issue with `getKAID()` (#217) it unfortunately cannot be done

Reply:
If you want to move this check, now that #217 is resolved, go ahead. I don't think it will cause issues though, so I might merge.

Reply:
Okay, I've moved it to a more suitable place now.